communication/mavlink_waypoint_handler_swarm.cpp

```diff
@@ -106,27 +107,25 @@ void Mavlink_waypoint_handler_swarm::set_circle_scenario(mavlink_command_long_t*
     waypoint_count_ = 2;
     current_waypoint_index_ = -1;
 
-    waypoint_transfo.origin = position_estimation_.local_position.origin;
-
     // Start waypoint
     uint32_t sysid = mavlink_stream_.sysid();
-    waypoint_transfo.pos[X] = circle_radius * cos(angle_step * (sysid - 1));
-    waypoint_transfo.pos[Y] = circle_radius * sin(angle_step * (sysid - 1));
-    waypoint_transfo.pos[Z] = altitude;
-    waypoint_global = coord_conventions_local_to_global_position(waypoint_transfo);
+    waypoint_transfo[X] = circle_radius * cos(angle_step * (sysid - 1));
+    waypoint_transfo[Y] = circle_radius * sin(angle_step * (sysid - 1));
+    waypoint_transfo[Z] = altitude;
+    coord_conventions_local_to_global_position(waypoint_transfo, ins_.origin(), waypoint_global);
```

Comment:
since you modified coord_conventions_local_to_global_position, shouldn't we use pointers ?